src/main/java/org/broadinstitute/consent/http/resources/DacResource.java

```diff
@@ -206,6 +213,36 @@ public Response filterUsers(@PathParam("term") String term) {
         return Response.ok().entity(users).build();
     }
 
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("{dacId}/dataset/{datasetId}")
+    @RolesAllowed({CHAIRPERSON})
+    public Response approveDataset(@Auth AuthUser authUser, @PathParam("dacId") Integer dacId, @PathParam("datasetId") Integer datasetId, String json) {
+        try{
+            User user = userService.findUserByEmail(authUser.getEmail());
+            Dataset dataset = datasetService.findDatasetById(datasetId);
+            if(dataset.getDacId() != dacId) { 
+                //Vague message is intentional, don't want to reveal too much info
+                throw new NotFoundException("Dataset not found");
+            }
+            userService.checkIfUserHasRole(UserRoles.CHAIRPERSON.getRoleName(), user, dacId);
```

Comment:
Just an idea, since you've already looked up the user, you could also use `User.hasUserRole()` and look for the `dacId` after that check. Alternatively, the same logic could also go in `User` since we're doing similar role-based checks there.